### DISCUSSION TITLE
Make the API cloud ready

### DISCUSSION THREAD

**Discussion by ArneVanH** [zenrosadira/abap-tbox-csvman#3] [Ideas]
Hi,

As mentioned in the previous post we just pulled your project and got it working on our system.  
And it works great, so thanks for that!

However... we are on a BTP - ABAP Cloud environment meaning we are restricted to the ABAP version for cloud.

This is a restricted version of ABAP, most notably denying access to all unreleased SAP-standard APIs (and thus most function-modules..)
Seeing as ABAP Cloud is the more restrictive of the two ABAP versions (and the direction SAP is headed with ABAP in general) it may be nice to start converting this API to be compatible with this version :)

See also: https://blogs.sap.com/2020/07/23/restricted-abap-and-sap-s-4hana-on-premise/

This will of course take quite some refactoring I guess... so not sure how actively you are still working on this project and have the time to continue to develop it :)  But just thought I'd throw out the request and see your reaction.

To give you an idea of the scope of the required changes, to get this API working on BTP we had to do the following:

> Create z-data elements and domains for all those not released for cloud (quite a few - SAP pushes you to define your own in ABAP cloud..)

> Remove the usage of the plausibility date & time checks

> Remove all conversion exit features

> Remove the 'get_ddic_field' methods

**Comment by zenrosadira** [comment]
I'm glad you are using this tool, and thanks for the tips!

I am working on general refactoring to make the whole code cloud ready while keeping as much functionality as possible

**Reply by ArneVanH** [reply]
In the meantime we've reimplemented the date and time checks using:

TRY.
        result = xco_cp_time=>date( iv_year  = date(4)
                                    iv_month = date+2(2)
                                    iv_day   = date+4(2) ).
      CATCH cx_no_check INTO DATA(exception).
        zcx_xco_cp_tm_date=>raise( textid = zcx_xco_cp_tm_date=>invalid_date
                                                       v1     = date ).
    ENDTRY.

similarly for time using: xco_cp_time=>time

This cloud API does the same checks as the 'check_plausibility' FMs

**Reply by zenrosadira** [reply]
Many thanks, I have integrated these methods into the code
